packages/lodestar/src/chain/chain.ts

```diff
@@ -411,4 +373,28 @@ export class BeaconChain extends (EventEmitter as { new(): ChainEventEmitter })
     this.logger.info(`Initializing beacon chain with eth1 block ${eth1Block.hash}`);
     await this.initializeBeaconChain(genesisState, merkleTree);
   };
+
```

Comment:
Can you add a description/motivation of the algorithm/problem here?

Reply:
sure